lib/rexml/source.rb

```diff
@@ -257,9 +257,11 @@ def read(term = nil, min_bytes = 1)
         rescue IOError
         end
         true
-      rescue Exception, NameError
+      rescue ArgumentError
         @source = nil
         false
+      rescue Exception, NameError
+        false
```

Comment:
Hmm. If we don't set `nil` to `@source`, `@source` isn't GC-ed, right?

Can we choose another approach something like the following?

```diff
diff --git a/lib/rexml/source.rb b/lib/rexml/source.rb
index 9950007..8b8ba0d 100644
--- a/lib/rexml/source.rb
+++ b/lib/rexml/source.rb
@@ -272,7 +272,7 @@ module REXML
         @scanner << readline(term)
       end
       if str
-        read if @scanner.eos? and !@source.eof?
+        read if @scanner.eos? and @source and !@source.eof?
         str
       else
         rest = @scanner.rest
```

BTW, can we add a test for this case like #286 does?

Reply:
> Can we choose another approach something like the following?

OK, I see.

> BTW, can we add a test for this case like https://github.com/ruby/rexml/pull/286 does?

OK, I see.